Update RELEASE-NOTES.md (#11197)
PrefectHQ/prefect: RELEASE-NOTES.md

## Release 2.14.5

### Storage block compatibility with `flow.from_source`

> You can now use all your existing storage blocks with `flow.from_source`! Using storage blocks with `from_source` is great when you need to synchronize your credentials and configuration for your code storage location with your flow run execution environments. Plus, because block configuration is stored serverside and pulled at execution time, you can update your code storage credentials and configuration without redeploying your flows!

Reviewer: ```suggestion
You can now use all your existing storage blocks with `flow.from_source`! Using storage blocks with `from_source` is great when you need to synchronize your credentials and configuration for your code storage location with your flow run execution environments. Plus, because block configuration is stored server-side and pulled at execution time, you can update your code storage credentials and configuration without re-deploying your flows!
```

Here's an example of loading and serving a flow from an S3 bucket:

```python
from prefect import flow
from prefect_aws import AwsCredentials
from prefect_aws.s3 import S3Bucket

if __name__ == "__main__":
    flow.from_source(
        source=S3Bucket(
            bucket_name="my-code-storage-bucket",
            credentials=AwsCredentials(
                aws_access_key_id="my-access-key-id",
                aws_secret_access_key="my-secret-access-key",
            ),
        ),
        entrypoint="flows.py:my_flow",
    ).deploy(name="my-deployment", work_pool_name="above-ground")
```

Here's an example of loading and deploying a flow from an S3 bucket:

```python
from prefect import flow
from prefect_aws.s3 import S3Bucket

if __name__ == "__main__":
    flow.from_source(
        source=S3Bucket.load("my-code-storage-bucket"), entrypoint="flows.py:my_flow"
    ).deploy(name="my-deployment", work_pool_name="above-ground")
```

> Note that a storage block must be saved before deploying a flow, but it doesn't need to be saved if you're serving a remotely stored flow.

Reviewer: ```suggestion
Note that a storage block must be saved before deploying a flow, but not if you're serving a remotely stored flow.
```
🤷‍♀️
